assets/src/stories-editor/store/middlewares.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import refx from 'refx';
+import multi from 'redux-multi';
+import { flowRight } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import effects from './effects';
+
+/**
+ * Applies the custom middlewares used specifically in the editor module.
+ *
+ * @param {Object} store Store Object.
+ *
+ * @return {Object} Update Store Object.
+ */
+function applyMiddlewares( store ) {
+	const middlewares = [
+		refx( effects ),
+		multi,
+	];
+
+	let enhancedDispatch = () => {
+		throw new Error(
+			'Dispatching while constructing your middleware is not allowed. ' +
+			'Other middleware would not be applied to this dispatch.'
+		);
+	};
+	let chain = [];
+
+	const middlewareAPI = {
+		getState: store.getState,
+		dispatch: ( ...args ) => enhancedDispatch( ...args ),
+	};
+	chain = middlewares.map( ( middleware ) => middleware( middlewareAPI ) );
+	enhancedDispatch = flowRight( ...chain )( store.dispatch );
```

Comment:
Why is the same variable `enhancedDispatch` used twice here, as the two definitions of it are wildly different and serve very different purposes? This code is pretty complex already, reusing this variable doesn't make it easier.

```suggestion
	const enhancedDispatch = flowRight( ...chain )( store.dispatch );
```

Reply:
I borrowed this from Gutenberg's middleware implementation for the most part 🤷‍♂ But your suggestion makes sense. I think it can be simplified in general.

Reply:
Hmm applying the suggestions actually breaks things, throwing `Dispatching while constructing your middleware is not allowed` errors in `assets/src/stories-editor/store/index.js`.

Reply:
Hm, that sounds weird. It shouldn't change anything?

Well, you can of course just ignore my suggestions. If it ain't broken and all that.

But maybe a comment would be helpful.